### DISCUSSION TITLE
Issue with Login Button Click (New thread)

### DISCUSSION THREAD

**Discussion by monojit18** [Sunbird-Ed/Community#227] [Installation] [answered]
Dear Sunbird team,

Starting this new thread from the earlier one (https://github.com/Sunbird-Ed/Community/discussions/213) as suggested by the team.
The Login button is not working for me even after successfully running PostInstall script.
I have also followed the thread - https://github.com/orgs/Sunbird-Ed/discussions/115 while running the PostInstall script and could resolve majority of the issues. But Login is still not working...and according to team, the indexing at ES is wrong!

So I would need some support/suggestion here!

I have NTP org created with channel=ntp and slug=ntp (DB extract below)
I have NTP org in ES as well (ES curl update below)

when I hit the following REST API, I am always getting 200 OK but count = 0
Can you please help/suggest, if I am still missing something?

========
http:///api/org/v1/search
{
"request": {
"filters": {
"id": ""
}
}
}

========
cqlsh> SELECT id,channel,slug from sunbird.organisation;

id | channel | slug
---------------------+-------------------+-------------------
0136386105328599041 | ArcheologyChannel | archeologychannel
ORG_001 | ROOT_ORG | null
0136444871414824966 | sunbird3 | sunbird3
0136443139879485445 | sunbird1 | sunbird1
0136443121580359684 | sunbird2 | sunbird2
0136445155885957127 | sunbird | sunbird
0136435154364579842 | SocialogyChannel | socialogychannel
0136445591068999680 | ntp | ntp

========
curl localhost:9200/_cat/indices?pretty
yellow open badgeassociations gJZT-m4NTaGoXlJY6bwW1g 5 1 0 0 1.2kb 1.2kb
yellow open cbatch-assessment PSQJ4ethQt2OtxFxdzZIpg 5 1 0 0 1.2kb 1.2kb
yellow open location 6O2pFF4aT7aDrnQ7_vuX7Q 5 1 6 0 62.7kb 62.7kb
yellow open course-batch mGux_PuXQTic4C-9L5YtkQ 5 1 0 0 1.2kb 1.2kb
yellow open userfeed 6fxgbJuYRciorlrec5_rlg 5 1 0 0 1.2kb 1.2kb
yellow open certreg pA3gyHWkTNmY8mQkqKUSdw 5 1 0 0 1.2kb 1.2kb
yellow open usercourses XGALLbdcQriKuX8xpwoK8Q 5 1 0 0 1.2kb 1.2kb
yellow open searchindex OeOGEBLfRoifhepALPLKxA 5 1 0 0 1.2kb 1.2kb
yellow open kp_audit_log_2022_38 C-g1-5tkR4KLj4mqoAxtwA 5 1 26 0 442.8kb 442.8kb
yellow open metrics SssqfaiNS6-eI-qH1DxVYQ 5 1 0 0 1.2kb 1.2kb
yellow open usernotes m3qdurNASkmxsIaftWcAdQ 5 1 0 0 1.2kb 1.2kb
yellow open dialcode bmD1C5nYQ4-KZd6qrKYP9A 5 1 0 0 1.2kb 1.2kb
yellow open dialcodemetrics VLkMLc-1R2y5XtG2yrcTdw 5 1 0 0 1.2kb 1.2kb
yellow open announcementtype oo-B3xHxSlKEGou-GBrwuA 5 1 0 0 1.2kb 1.2kb
yellow open compositesearch 0jXKAPGXT2-X45bdxbCA-w 5 1 0 0 1.2kb 1.2kb
yellow open cbatch ObkwmN4xTSOYySwSli3QVg 5 1 0 0 1.2kb 1.2kb
yellow open user_alias HfUd2rSMQjW3ineuL2bbPA 5 1 6 0 88.7kb 88.7kb
yellow open user-courses JLitTF_vQ62ITPgiwMFDxw 5 1 0 0 1.2kb 1.2kb
yellow open userprofilevisibility qJqH-gHUTv-99bjXJhbH-A 5 1 0 0 1.2kb 1.2kb
yellow open org_alias Tb2I3J6yRKaDaGv8CdAxsw 5 1 7 0 70.3kb 70.3kb
yellow open cert-templates J0Ti5QyfTwSIISyLQYnWjQ 5 1 0 0 1.2kb 1.2kb
yellow open announcement PcXpZW7RTI-41kdWYiSs6Q 5 1 0 0 1.2kb 1.2kb
yellow open cbatchstats adTuTLG7QfW95z0zGtF9NA 5 1 0 0 1.2kb 1.2kb

========
curl -sS -X GET "localhost:9200/org_alias/_search?pretty" -H 'Content-Type: application/json' -d '{

"query": {
    "match" : {
        "slug" : "ntp"
    }
}
}'
{
"took" : 1,
"timed_out" : false,
"_shards" : {
"total" : 5,
"successful" : 5,
"skipped" : 0,
"failed" : 0
},
"hits" : {
"total" : 1,
"max_score" : 0.2876821,
"hits" : [
{
"_index" : "org_alias",
"_type" : "_doc",
"_id" : "0136445591068999680",
"_score" : 0.2876821,
"_source" : {
"id" : "0136445591068999680",
"channel" : "ntp",
"createdBy" : "ad1ba6b9-9864-41ef-972e-6add52b7ed66",
"createdDate" : "2022-10-12 17:10:16:799+0000",
"description" : "NTP Organisation for Sunbird",
"orgName" : "NTP",
"organisationType" : 5,
"rootOrgId" : "0136445591068999680",
"slug" : "ntp",
"status" : 1,
"isSSOEnabled" : false,
"isTenant" : true,
"orgLocation" : [ ],
"identifier" : "0136445591068999680"
}
}
]
}
}

========
curl -sS -X GET "localhost:9200/org_alias/_search?pretty" -H 'Content-Type: application/json' -d '{

"query": {
    "match" : {
        "slug" : "sunbird"
    }
}
}'
{
"took" : 1,
"timed_out" : false,
"_shards" : {
"total" : 5,
"successful" : 5,
"skipped" : 0,
"failed" : 0
},
"hits" : {
"total" : 1,
"max_score" : 0.2876821,
"hits" : [
{
"_index" : "org_alias",
"_type" : "_doc",
"_id" : "0136445155885957127",
"_score" : 0.2876821,
"_source" : {
"id" : "0136445155885957127",
"channel" : "sunbird",
"createdBy" : "ad1ba6b9-9864-41ef-972e-6add52b7ed66",
"createdDate" : "2022-10-12 15:50:20:493+0000",
"description" : "Default Channel",
"orgName" : "Sunbird",
"organisationType" : 5,
"rootOrgId" : "0136445155885957127",
"slug" : "sunbird",
"status" : 1,
"isSSOEnabled" : false,
"isTenant" : true,
"orgLocation" : [ ],
"identifier" : "0136445155885957127"
}
}
]
}
}

**Comment by monojit18** [comment]
Dear Team,

Any suggestion/guidance on this?

**Reply by beepdot** [reply]
Hello @monojit18 I am going to provide you few high level instructions on how to resolve this. You should be easily be able to find the APIs or commands from the internet. The problem here is we should be seeing `org` and `user` indices as their names and then these indices should have alias as `org_alias` or `user_alias` attached to them.
Based on your release version, you should be able to find what version of the index we should be seeing. For example, in the `master` branch we can see it as `orgv3` and `userv3` - https://github.com/project-sunbird/sunbird-devops/tree/master/ansible/roles/es-mapping/files/indices

#### Steps to resolve this:
1. Create the indices manually or by using the correct version of Jenkins job. If you are doing it manually, you can use the index and mapping file to create them. You can also talk a look at the `es-mapping` ansible role to see how it creates. Please look at the `indices` and `mappings` folder here - https://github.com/project-sunbird/sunbird-devops/tree/master/ansible/roles/es-mapping/files
2. Once you have the index created, you can use the Elasticsearch reindex api to reindex the data from your current `org_alias` index to `orgv3` and similar way for user index.
3. Once the data is reindexed, you can delete your old indices `org_alias` and `user_alias`
4. Add an index alias to your new indices. Add alias `org_alias` to `orgv3` and alias `user_alias` to `userv3`

Some links for your reference
https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-reindex.html
https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-add-alias.html

I am posting a sample from our environment
#### Indices
```
curl -sS localhost:9200/_cat/indices | egrep "orgv3|userv3"
green open orgv3                              rqK90yPbRymZj7W3e3j6gA 5 1  802567      4    1.4gb  734.2mb
green open userv3                             7kc6JSwRQl6IuMBHxCcNWg 5 1   61297   6535  285.6mb  143.3mb
```

#### Aliases
```
curl -sS localhost:9200/orgv3 | jq .orgv3.aliases
{
  "org_alias": {}
}

curl -sS localhost:9200/userv3 | jq .userv3.aliases
{
  "user_alias": {}
}
```

**Comment by sharathkashyap** [comment, marked as answer]
@monojit18 Can you please let us know if the issue is resolved.

**Reply by chitranshu-keshav** [reply]
@monojit18 Reminder to mark this thread as answered if your issue is resolved
